Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py

```diff
@@ -0,0 +1,766 @@
+"""
```

Comment:
It is fairly difficult to spot the differences between this file and the one it is moved from. Are there any areas with meaningful changes you want reviewers to focus on?

Reply:
The reviews you have been giving regarding naming and what attributes should go where (`__init__` vs. non-`__init__` etc.) are sufficient. Ideally there will be another task later on to tidy this up to make it look more neat/clean, but this one primarily deals with extending the support. Also if you see anything that you think could go into `multi_test_framework.py` that should be moved out from the inheriting classes would also be good review feedback.

Reply:
Also just running some tests using this code to make sure no unintended side effects are observed on the user's end.